wordpress-mobile/WordPress-Android
# [Gutenberg] Add blog ID to editor session events

Following suit with https://github.com/wordpress-mobile/WordPress-iOS/pull/17085, this PR adds the `blog_id` to the editor session events.

This is done essentially by reusing the [AnalyticsUtils.trackWithSiteDetails()](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/util/analytics/AnalyticsUtils.java#L243) helper method.

All editor session events have been augmented with this PR:
* `wpandroid_editor_session_start`
* `wpandroid_editor_session_end`
* `wpandroid_editor_session_switch_editor`
* `wpandroid_editor_session_template_apply`

Note: I had to refactor some of the Analytics code to allow for passing around `AnalyticsTrackerWrapper` to make the methods testable.

### To test

1. Open the block editor.
1. Tap the three-dot menu in the top right.
1. Tap Switch to HTML Mode.
1. Tap the three-dot menu in the top right.
1. Tap Switch to Visual Mode.
1. Close the editor.
1. Verify the following Tracks events include the correct `blog_id`
   property
    * `wpandroid_editor_session_start`
    * `wpandroid_editor_session_switch_editor`
    * `wpandroid_editor_session_end`

### To test the automated tests
1. Comment out the `blog_id` property insertion [at the "source"](https://github.com/wordpress-mobile/WordPress-Android/blob/96b3eecd3145f9f5463620406507c2ef82d1e620/WordPress/src/main/java/org/wordpress/android/util/analytics/AnalyticsUtils.java#L262)
2. Run the `PostEditorAnalyticsSessionTest` via AndroidStudio or in the CLI via `./gradlew :WordPress:cleanTestWordpressWasabiDebugUnitTest :WordPress:testWordpressWasabiDebugUnitTest --tests "org.wordpress.android.ui.posts.editor.PostEditorAnalyticsSessionTest"`
3. Notice the tests failing with `Argument(s) are different! Wanted:` error where the blog_id is entirely missing from the `Actual invocations have different arguments:` parts.

Another way to test is to:

1. Modify `MapHasEntry("blog_id" to blog_id)` to something like `MapHasEntry("blog_id" to 2222L)`
2. Run the tests as before and notice them error with `Argument(s) are different!` where the actual invocations include `1234L` but the test waiting for `2222L`.
3. 

## Regression Notes
1. Potential unintended areas of impact
As I had to refactor the `AnalyticsUtils.trackWithSiteDetails()` method a bit (to make it more testable via `AnalyticsTrackerWrapper`), tracks that use that method may be impacted if buggy.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

I tried sharing a photo to WP starting from the camera app, and that triggered the `wpandroid_share_to_wp_succeeded` event, which I verified appearing OK in the backend.

3. What automated tests I added (or what prevented me from doing so)
Added tests that verify that the tracks have the `blog_id` property.

PR submission checklist:
